django-stubs/db/models/fields/json.pyi

```diff
@@ -29,8 +29,8 @@ class JSONField(CheckFieldDefaultMixin, Field[Union[_A, Combinable], _A]):
     def get_transform(self, name: Any) -> Any: ...
     def value_to_string(self, obj: Any) -> Any: ...
     @overload
-    def __new__(  # type: ignore [misc]
-        cls,
+    def __init__(
+        self: JSONField[_A],
```

Comment:
I dont believe an annotation is needed for `self`. At least I never see it in typeshed repo

Reply:
I think the annotation is required here to handle the Optional case. You can find examples of this in the typeshed repository too:

https://cs.github.com/python/typeshed/blob/d1375f600cc4a7e7d81dcd9229c6abb933efda2a/stdlib/array.pyi?q=__init__%28self%3A#L24-L28